crates/crypto/src/key/type/ecdsa/k256.rs

```diff
@@ -0,0 +1,295 @@
+use crate::{
+    key::{KeyConversionError, KeyMetadata},
+    AlgorithmInstance, Error, KeyType, PublicKey, RecoveryKey, RejectedSignature, SecretKey,
+    SignatureVerification, SigningKey, VerifyingKey,
+};
+pub use k256::ecdsa::{SigningKey as K256SecretKey, VerifyingKey as K256PublicKey};
+use sha2::Digest;
+
+use super::{EcdsaKeyType, EcdsaPublicKey, EcdsaSecretKey};
+
+impl PublicKey {
+    /// Creates a new ECDSA K-256 public key.
+    pub fn new_ecdsa_k256(x: &[u8], y: &[u8]) -> Result<Self, KeyConversionError> {
+        EcdsaPublicKey::new_k256(x, y).map(Self::Ecdsa)
+    }
+
+    /// Decodes an ECDSA P-256 [`PublicKey`] (compressed or uncompressed) from
+    /// the `Elliptic-Curve-Point-to-Octet-String` encoding described in
+    /// SEC 1: Elliptic Curve Cryptography (Version 2.0) section
+    /// 2.3.3 (page 10).
+    ///
+    /// See: <http://www.secg.org/sec1-v2.pdf>
+    pub fn from_ecdsa_k256_sec1_bytes(bytes: &[u8]) -> Result<Self, KeyConversionError> {
+        EcdsaPublicKey::from_k256_sec1_bytes(bytes).map(Self::Ecdsa)
+    }
+}
+
+impl EcdsaPublicKey {
+    /// Creates a new ECDSA K-256 public key.
+    pub fn new_k256(x: &[u8], y: &[u8]) -> Result<Self, KeyConversionError> {
+        let mut bytes = Vec::new();
+        bytes.push(0x04);
+        bytes.extend(x);
+        bytes.extend(y);
+        Self::from_k256_sec1_bytes(&bytes)
```

Comment:
FWIW you can use `k256::EncodedPoint::from_untagged_bytes` or `from_affine_coordinates` to avoid messing around with SEC1 tags.

Reply:
Most of the key conversion functions are coming straight from `ssi_jwk`.

> avoid messing around with SEC1 tags

Oh that's what the `0x04` is? Yeah that's why I did change anything, I wasn't sure what format that was.

Reply:
Oh fair enough, it's the same in the end anyway, was more of an FYI